Luke7787/ImmaculateInventors
# Feature/get user items

Summary:
- get items specific to a user
- get items by name

How to test:
-npx nodemon backend.tsx or node backend.tsx

In boomerang, test to see if you can get items specific to a user:
(post)http://localhost:8000/users               <----adds a user to the database
{
    "username": "BIGBOY",
    "password": "Password123!"                   <----make sure to include this in the body, the username is up to you, the password 
                                                                            you put needs to meet the requirements
}

If successful, it should print something like this on the right:
{
    "username": "BIGBOY",
    "password": "Password123!",
    "items": [],
    "_id": "65ee5161ee4d6adcc263d9a1",
    "__v": 0
}

Now add a few items to the user:
(patch)http://localhost:8000/itemToUser?uid=65ee5161ee4d6adcc263d9a1         <----uid is the id of the user
{
    "userId": "65d91ec271d8d3c63dcc7a97",            <----the same uid
    "name": "pentagon",                                            <----name of the item
    "quantity": 3                                                         <----the quantity the item is initialized with
}

Now see if you get the items from that same user:
(get)http://localhost:8000/items?uid=65d91e81dd5362a0bcd237a9                     <----the same uid 

If successful, it should print something like this on the right:
[
{
    "userId": "65d91ec271d8d3c63dcc7a97",
    "name": "pentagon",
    "quantity": 3
}
]

Try adding another user and add an item to that user, then try getting items from that user and it should only print items associated to that user and not all the items in the database


Now test to see if you can get an item from the database by its name:
(get)http://localhost:8000/items?itemName=pentagon                              <----name of the item

If successful, it should print something like this on the right:
[
    {
        "_id": "65eb9802b566ba4fa9984f4a",
        "userId": "65d91ec271d8d3c63dcc7a97",
        "name": "pentagon",
        "quantity": 3,
        "__v": 0
    }
]
